src/Fantasma/Internal/Storage/EntityFramework/EntityFrameworkStorage.cs

```diff
@@ -0,0 +1,141 @@
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using JsonSerializer = System.Text.Json.JsonSerializer;
+
+namespace Fantasma.Internal;
+
+internal sealed class EntityFrameworkStorage : IJobStorage
+{
+    private readonly TimeProvider _time;
+    private readonly IServiceProvider _provider;
+    private readonly ILogger<EntityFrameworkStorage> _logger;
+    private readonly IServiceScope _scope;
+
+    public EntityFrameworkStorage(
+        TimeProvider time,
+        IServiceProvider provider,
+        ILogger<EntityFrameworkStorage> logger)
+    {
+        ArgumentNullException.ThrowIfNull(provider);
+
+        _time = time ?? throw new ArgumentNullException(nameof(time));
+        _provider = provider ?? throw new ArgumentNullException(nameof(provider));
+        _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        _scope = provider.CreateScope();
+    }
+
+    public void Dispose()
+    {
+    }
+
+    public async Task Add(Job job)
+    {
+        using var scope = _provider.CreateScope();
+        using var db = scope.ServiceProvider.GetRequiredService<IFantasmaDatabase>();
+
+        if (await db.Jobs.FindAsync(job.Id) != null)
+        {
+            return;
+        }
+
+        db.Jobs.Add(
+            new FantasmaJob
+                {
+                    Id = job.Id,
+                    ClrType = job.Data.GetType().AssemblyQualifiedName ?? throw new InvalidOperationException(),
+                    Data = JsonConvert.SerializeObject(job.Data),
+                    ScheduledAt = job.ScheduledAt.UtcDateTime,
+                    Status = JobStatus.Scheduled,
+                    Cron = job.Cron,
+                });
+
+        await db.SaveChangesAsync();
+    }
+
+    public async Task Remove(Job job)
+    {
+        using var scope = _provider.CreateScope();
+        using var db = scope.ServiceProvider.GetRequiredService<IFantasmaDatabase>();
+
+        var dbJob = await db.Jobs.FindAsync(job.Id);
+        if (dbJob != null)
+        {
+            db.Remove(dbJob);
+            await db.SaveChangesAsync();
+        }
+    }
+
+    public async Task Update(Job job)
+    {
+        using var scope = _provider.CreateScope();
+        using var db = scope.ServiceProvider.GetRequiredService<IFantasmaDatabase>();
+
+        var dbJob = await db.Jobs.FindAsync(job.Id);
+        if (dbJob != null)
+        {
+            dbJob.Status = job.Status;
+            dbJob.ScheduledAt = job.ScheduledAt.UtcDateTime;
+            await db.SaveChangesAsync();
+        }
+    }
+
+    public Task<Job?> GetNextJob()
+    {
+        using var scope = _provider.CreateScope();
+        using var db = scope.ServiceProvider.GetRequiredService<IFantasmaDatabase>();
+
+        var now = _time.GetUtcNow();
+        var nextJob = db.Jobs.Where(x => x.ScheduledAt < now).AsEnumerable().MinBy(x => x.ScheduledAt);
```

Comment:
```suggestion
        var nextJob = db.Jobs.Where(x => x.ScheduledAt < now).AsNoTracking().AsEnumerable().MinBy(x => x.ScheduledAt);
```

Since Jobs aren't updated here, they can be loaded without change tracking info